realm/realm-library/src/androidTest/java/io/realm/TypeBasedNotificationsTest.java

```diff
@@ -0,0 +1,1367 @@
+/*
+ * Copyright 2015 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm;
+
+import android.os.Handler;
+import android.os.HandlerThread;
+import android.test.AndroidTestCase;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import io.realm.entities.Dog;
+import io.realm.proxy.HandlerProxy;
+
+public class TypeBasedNotificationsTest extends AndroidTestCase {
+    private HandlerThread handlerThread;
+    private Handler handler;
+    private CountDownLatch signalTestFinished;
+    private AtomicInteger globalCommitInvocations;
+    private AtomicInteger fineGrainedCommitInvocations;
+    private RealmConfiguration configuration;
+    private Realm realm;
+
+    @Override
+    protected void setUp() throws Exception {
+        handlerThread = new HandlerThread("LooperThread");
+        handlerThread.start();
+        handler = new Handler(handlerThread.getLooper());
+        signalTestFinished = new CountDownLatch(1);
+        globalCommitInvocations = new AtomicInteger(0);
+        fineGrainedCommitInvocations = new AtomicInteger(0);
+        configuration = new RealmConfiguration.Builder(getContext()).build();
+        Realm.deleteRealm(configuration);
+    }
+
+    @Override
+    protected void tearDown() throws Exception {
+        final CountDownLatch cleanup = new CountDownLatch(1);
+        if (realm != null && handler.getLooper().getThread().isAlive()) {
+            handler.post(new Runnable() {
+                @Override
+                public void run() {
+                    if (!realm.isClosed()) {
+                        realm.close();
+                    }
+                    Realm.deleteRealm(configuration);
+                    handlerThread.quit();
+                    realm = null;
+                    cleanup.countDown();
+                }
+            });
+            TestHelper.awaitOrFail(cleanup);
+        }
+    }
+
+    // ********************************************************************************* //
+    // UC 1.
+    // Callback should be invoked after a relevant commit (one that should impact the
+    // query from which we obtained our RealmObject or RealmResults)
+    // ********************************************************************************* //
+    //FIXME failing because sync RealmObject doesn't have the initial currentVersion, so any first commit
```

Comment:
The unit test isn't failing. So isn't this more like a TODO that hopefully will be fixed with proper fine-grained support?